### DISCUSSION TITLE
Positive feedback!

### DISCUSSION THREAD

**Discussion by rbnpi** [ExistentialAudio/BlackHole#29] [General]
Just came across this and tried on my Mac with Sonic Pi and Helm synthesiser. Worked great for feeding Helm audio back into Sonic Pi input. Many thanks. Better performance than SoundFlower and saves the huge cost of Amoeba's LoopBack!

**Comment by devinroth** [comment]
Thanks! Glad it's working well for you.

**Comment by oheyjesse** [comment]
Hi @rbnpi - I'm here for similar reasons - Except I'm trying to route SonicPi's output audio through and back into my device for screen recordings - I'd basically like to record the audio output of SonicPi while I record it using quicktime or the like. Have you had any luck doing something like this? 

I'd like to route SonicPi's audio output into blackhole, but I can't find the setting / functionality to do so.

**Comment by rbnpi** [comment]
To record SP in QuickTime do this.

Create an aggregate output device in Audio-Midi Setup, and select BlackHole and Built-IN Output as Output devices. Click drift correction for BlackHole.
Set the Aggregate Output device as default output.
Set BlackHole as Default Input.

Having set this up, start Sonic Pi. If you play something you should be able to hear it in your Mac Speakers.
If you also start QuickTime as  screen recorder, you can select Black Hole as the input device.
You should see the audio “meter” in Quick Time responding to Audio from Sonic Pi and if you record you will get the screen picture with Sonic Pi audio.
NB when using the aggregate device the Mac volume control will be ineffective.

Hope this works for you.

**Comment by oheyjesse** [comment]
Hey @rbnpi - Thanks very much for the tips! You're spot on, it worked well for me when I poked it some more. Appreciate you taking the time to sort it out for me.

It seemed my issue was that in my Audio Midi Setup for the Multi Output Device, I needed to mess around with it somewhat. I'm running Catalina and a few things seem slightly different- I had to select my headphone output as the primary (it didn't seem to work with BlackHole as the default, for me?), and tick and untick the blackhole device from the list of devices, and mess around with the drift correction checkboxes too. After messing with these a little, it sprang to life - Previously I'd not been getting any passthrough of audio through Blackhole in this manner.

**Comment by rbnpi** [comment]
I’m keeping clear of Catalina at present for reasons such as these! Glad you got it working.

**Comment by james-rae** [comment]
Was crying tears when Soundflower stopped working for me years ago. Tears have dried! Beauty bit of software.

**Comment by quantuminformation** [comment]
I have had a world of pain with ISHOWYOU audio, now trying this..

**Comment by quantuminformation** [comment]
Made an ep about BH

https://learnwithus.simplecast.com/episodes/devin-roth?fbclid=IwAR2Y4lFEmBl_pyDA6kxn6j-CJQ09c6ZN6EPHGaaOGj5kMJn04SlC9wBoHes
